### DISCUSSION TITLE
IfcOpenShell not built for 'windows\64bit\python3.11'

### DISCUSSION THREAD

**Discussion by SajjadNia** [IfcOpenShell/IfcOpenShell#2714] [Q&A]
hey there, I want to use ifc open shell in my script but I am faced with this error.
is it possible that because of my 3.11 version?
(I have to mention i installs ifc open shell for python version 11)

![image](https://user-images.githubusercontent.com/122394275/213916786-92f0435e-ba24-4cbb-b240-a93120b4edc8.png)

thank you in advance

**Comment by someparsa** [comment]
This shows that the IFC python bindings are compatible with lower versions of the python. Not always the most updated versions are suitable to be used with all packages. To the best of my understanding, version 3.10 works good with this library. You can install lower versions alongside any other version you have on your system.

**Comment by expertegorov** [comment]
Had the same problem.
The solution helped me.
Remove installed version:
pip uninstall ifcopenshell
Download pre-built package "py311-win64" from: https://blenderbim.org/docs-python/ifcopenshell-python/installation.html
Unpack and copy folder "ifcopenshell" to path where is installed python: c:\Users\USERNAME\AppData\Local\Programs\Python\Python311\Lib\site-packages\
Then comand "import ifcopenshell" works.

**Reply by someparsa** [reply]
I wonder the prerequisites and programming back-end would work properly by this direct manipulation of the root folders. As long as the codes and algorithms are packaged with an earlier version of the translator. Anyway, I hope this works.
Another solution may be using temporary environments which successfully creates different definitions of the python on the system and then each version can have its own library ports. And the easiest solution can be always stepping back from an on edge versioning and using one lower python version on the system.

**Reply by expertegorov** [reply]
Works well.
I just used the installation instructions provided on the website: "Unzip the downloaded file and copy the ifcopenshell directory into your Python path."

**Comment by Moult** [comment]
@SajjadNia how did you install IfcOpenShell? Did you follow one of the instructions from https://blenderbim.org/docs-python/ifcopenshell-python/installation.html ? If so, which? What environment is this?
